public/static/js/init.py

```diff
@@ -799,29 +811,48 @@ def run_turtle_cmd(msg):
 def hit_breakpoint(lineno, alocals, aglobals, is_secondary=False):
     global step_into
     global last_seen_lineno
+    global watches
+    global in_watch
     if is_secondary and last_seen_lineno == lineno:
         return True
     last_seen_lineno = lineno
-    if step_into or lineno in active_breakpoints:
+    if not in_watch and (step_into or lineno in active_breakpoints):
```

Comment:
if currently executing a watched expression, then don't trigger a breakpoint (otherwise it could result in recursive breaking)